js/CommandInterface.js

```diff
@@ -0,0 +1,347 @@
+                const option = document.createElement("option");
+                option.innerText = name;
+                option.setAttribute("value", name);
+                if(name == "copy"){
+                    option.toggleAttribute("selected");
+                    // copy takes no args
+                    editor.toggleAttribute("disabled");
+                    editor.setAttribute("Placeholder", command_registry[name].description)
+                }
+                selection.appendChild(option);
+            });
+            // add a close button
+            const svg = createIconSVGFromString(icons.circleX);
+            const button = document.createElement("span");
+            button.id = "close";
+            button.appendChild(svg);
+            button.addEventListener("click", this.onClose);
+            header.appendChild(button);
+            // add events listeners for the buttons
+            const seeit_button = this.shadowRoot.querySelector("button#see-it");
+            const doit_button = this.shadowRoot.querySelector("button#do-it");
+            const saveit_button = this.shadowRoot.querySelector("button#save-it");
+            // TODO these should be proper tooltips not titles
+            seeit_button.addEventListener("mousedown", this.onSeeIt);
+            seeit_button.addEventListener("mouseup", this.onSawIt);
+            seeit_button.setAttribute("title", "See what will happen before you commit!");
+            doit_button.addEventListener("click", this.onDoIt);
+            doit_button.setAttribute("title", "Save it before you do it!");
+            saveit_button.addEventListener("click", this.onSaveIt);
+            saveit_button.setAttribute("title", "Save it to the program");
+        }
+    }
+
+    disconnectedCallback() {
+        const header = this.shadowRoot.querySelector("#header");
+        const footer = this.shadowRoot.querySelector("#footer");
+        const selection = header.querySelector("#available-commands");
+        header.removeEventListener("mousedown", this.onMouseDownInHeaderFooter);
+        footer.removeEventListener("mousedown", this.onMouseDownInHeaderFooter);
+        selection.removeEventListener("change", this.onComamndSelection);
+        const seeit_button = this.shadowRoot.querySelector("button#see-it");
+        seeit_button.removeEventListener("mousedown", this.onSeeIt);
+        seeit_button.removeEventListener("mouseup", this.onSawIt);
+        const doit_button = this.shadowRoot.querySelector("button#do-it");
+        doit_button.removeEventListener("click", this.onDoIt);
+        const saveit_button = this.shadowRoot.querySelector("button#save-it");
+        saveit_button.removeEventListener("click", this.onSaveIt);
+    }
+
+
+    onComamndSelection(event){
+        const name = event.target.value;
+        const editor = this.shadowRoot.querySelector("#editor");
+        const command_info = this.interpreter.command_registry[name];
```

Comment:
There seems to be a mix in here of snake_case and camelCase. Should we update to stick with camelCase? (maybe you were switching between this and a python project)

Reply:
fixed [here](https://github.com/dkrasner/worksheet/pull/35/commits/a93651474ccb3e1d8328d8288e38eedd044df09a)